sympy/combinatorics/free_group.py

```diff
@@ -0,0 +1,938 @@
+from __future__ import print_function, division
+
+from sympy.core.basic import Basic
+from sympy.core.compatibility import is_sequence, as_int, string_types
+from sympy.core.expr import Expr
+from sympy.core.symbol import Symbol, symbols as _symbols
+from sympy.core.sympify import CantSympify
+from mpmath import isint
+from sympy.core import S
+from sympy.printing.defaults import DefaultPrinting
+from sympy.utilities import public
+from sympy.utilities.iterables import flatten
+from sympy.utilities.magic import pollute
+
+
+@public
+def free_group(symbols):
+    """Construct a free group returning ``(FreeGroup, (f_0, f_1, ..., f_(n-1))``.
+
+    Parameters
+    ----------
+    symbols : str, Symbol/Expr or sequence of str, Symbol/Expr (may be empty)
+
+    Examples
+    ========
+
+    >>> from sympy.combinatorics.free_group import free_group
+    >>> F, x, y, z = free_group("x, y, z")
+    >>> F
+    <free group on the generators (x, y, z)>
+    >>> x**2*y**-1
+    x**2*y**-1
+    >>> type(_)
+    <class 'sympy.combinatorics.free_group.FreeGroupElm'>
+
+    """
+    _free_group = FreeGroup(symbols)
+    return (_free_group,) + tuple(_free_group.generators)
+
+@public
+def xfree_group(symbols):
```

Comment:
specifically the `xfree_group` could be used for collecting the generators of free group in a single variable, directly. Similarly the next function `vfree_group` could be use to pollute the environment with the generator variables automatically. I don't say we can't live without it. But its good to have it.